src/org/exist/collections/Collection.java

```diff
@@ -1096,6 +1096,20 @@ public void removeCollection(final DBBroker broker, final XmldbURI name) throws
         }
     }
 
+    public void removeResource(Txn txn, DBBroker broker, DocumentImpl doc) throws PermissionDeniedException, LockException, IOException, TriggerException {
+        if (doc == null) return;
+
+        if (doc.getCollection() != this) {
```

Comment:
@shabanovd I have no idea how what you just said is related to the change I asked for!